spec/integ/admin-rooms.spec.js

```diff
@@ -543,4 +543,113 @@ describe("Admin rooms", function() {
         expect(createdMatrixRoom).toBe(true, "Did not create matrix room");
         expect(joinedChannel).toBe(true, "Bot didn't join channel");
     }));
+
+    it("should allow arbitrary IRC commands to be issued",
+    test.coroutine(function*() {
+        var newChannel = "#coffee";
+
+        var cmdCount = 0;
+        env.ircMock._whenClient(roomMapping.server, userIdNick, "send",
+        function(client) {
+            cmdCount++;
+        });
+
+        // 5 commands should be executed
+        // rubbishserver should throw and prevent more commands
+        let command = `${roomMapping.server}\n` +
+                        `JOIN ${newChannel}\n` +
+                        `TOPIC ${newChannel} :some new fancy topic\n` +
+                        `PART ${newChannel}\n` +
+                        `STUPID COMMANDS\n` +
+                        `${roomMapping.server}\n` +
+                        `SOME COMMAND\n` +
+                        `rubbishserver\n` +
+                        `PART ${newChannel}`;
```

Comment:
You should strip out the \n and do it as a big \` block (it'll test that `trim()` function is working *and* look nicer :smile:)